tests/std/tests/P0896R4_views_common/test.cpp

```diff
@@ -63,6 +63,46 @@ void non_literal_parts(R& r, E& expected) {
             }
         }
     }
+
+#if _HAS_CXX23
+    using ranges::const_iterator_t;
+
+    const same_as<const_iterator_t<R>> auto cfirst = r.cbegin();
+    if (!is_empty) {
+        assert(*cfirst == *begin(expected));
+    }
+
+    if constexpr (copyable<V>) {
+        auto r2                                         = r;
+        const same_as<const_iterator_t<R>> auto cfirst2 = r2.cbegin();
+        if (!is_empty) {
+            assert(*cfirst2 == *cfirst);
+        }
+    }
+
+    if constexpr (CanCBegin<const R&>) {
+        const same_as<const_iterator_t<const R>> auto cfirst3 = as_const(r).cbegin();
+        if (!is_empty) {
+            assert(*cfirst3 == *cfirst);
+        }
+    }
+
+    const same_as<const_iterator_t<R>> auto clast = r.cend();
+    if constexpr (bidirectional_iterator<R>) {

```

Comment:
This says `bidirectional_iterator` but I am virtually certain that `bidirectional_range` (as on line 99) was intended.